cmd/airflow.go

```diff
@@ -150,6 +157,8 @@ func newAirflowInitCmd() *cobra.Command {
 	}
 	cmd.Flags().StringVarP(&projectName, "name", "n", "", "Name of Astro project")
 	cmd.Flags().StringVarP(&airflowVersion, "airflow-version", "a", "", "Version of Airflow you want to create an Astro project with. If not specified, latest is assumed. You can change this version in your Dockerfile at any time.")
+	cmd.Flags().StringVarP(&fromTemplate, "from-template", "t", "", "Template name that you want to use to create the local astro project. Possible values can be etl, dbt-on-astro, generative-ai and learning-airflow. Please note template based astro project use latest runtime version and airflow-version flag will be ignored when creating a project with template flag")
+	cmd.Flags().BoolVarP(&selectTemplate, "select-template", "s", false, "Provides a list of templates to select from and create the local astro project based on selected template")
```

Comment:
I quite like this UX, the user is presented with the valid choices and picks the one they want. Can we just make this the only way to create a project from a template? It could even be what `--from-template` does and then we don't need `--select-template`. We shouldn't need a non-interactive variant because this isn't going to be called by scripts/CI/etc.

Reply:
Hey @jeremybeard ,

The initial Idea was to provide users with simple set of commands that they can execute to set up template based astro project. Eg: `astro dev init --from-template etl` 

If we keep only `--select-template` flag functionality then we will have to guide users in our onboarding workflow to select the template and set up astro project accordingly. 

CC: @schnie

Reply:
@pritt20 we could use the same flag _if_ we used the `=` format right? 

So `astro dev init --from-template` would be interactive, and
`astro dev init --from-template=etl` would work for the one-shot version?

Reply:
yeah, this format can be implemented to use same flag for interactive and for one-shot version. Its just that we don't really use this format anywhere in CLI for existing commands😅

Reply:
Yea that does make it a little awkward I suppose. The main use-case for the one-shot style will be to copy and paste from the UI, so maybe that won't matter too much. 🤔

Reply:
I kind of lean towards using a single flag and just accept that we need to show the `=` in the commands we display to copy/paste.

Reply:
Hey @jeremybeard @neel-astro, 

Could you please let us know your thoughts on above discussion. If we want to use single flag and use it both ways, shall we go ahead with the format `astro dev init --from-template=etl` ?

Reply:
From a CLI UX perspective, I might be a bit concerned about changing the behavior of a single flag in the CLI, where a user could only pass a flag value with `=`.
From what I understand based on comments if passing a template value in the command isn't going to be a major form of usage (because the user has to copy/paste the value from somewhere), then should we just skip that and always present an interactive mode? . `astro dev init` does not have a CI-CD use case, so we don't necessarily need to support a non-interactive mode.

Reply:
Our next steps for the onboarding flow are to show the user a small block of commands to copy and execute in their terminal. I think we want a non-interactive way to do this to reduce the keystrokes (minor, I know) and steps involved to get setup. It'd be great to just paste in the terminal and go with no more interactions. 

Given that, we have a choice to either piggyback on the same flag (`--from-template`), or add a separate, but similar command. The latter feels a little awkward and a little confusing. For that reason, I lean towards just using the `=` format for this command alone right now, but keep our existing pattern for other flags. We can treat this as a special case for onboarding purposes. 

I imagine most users exploring the CLI on their own would run the command interactively. For our onboarding use-case we want to specify the exact template to use, so we'd provide them a command to run that matches the template they selected when going through the onboarding flow, like `astro dev init --from-template=generative-ai`.

Maybe we just document the bare, interactive `--from-template` mode and reserve the `=` format (`--from-template=generative-ai` just for the case where we give them the command to copy/paste.

Reply:
> Maybe we just document the bare, interactive --from-template mode and reserve the = format (--from-template=generative-ai just for the case where we give them the command to copy/paste.

That sounds good to me.

Reply:
> Our next steps for the onboarding flow are to show the user a small block of commands to copy and execute in their terminal.

Cool, I was missing this context, I am on board with that approach 👍

Reply:
Thanks everyone! I've updated the PR to reflect above discussed changes.